hack/cnbi-package-list.yaml

```diff
@@ -0,0 +1,193 @@
+    - name: create-image-stream
+      taskRef:
+        name: openshift-client
+        kind: ClusterTask
+      params:
+        - name: SCRIPT
+          value: |
+            cat <<EOM | oc apply -f -
+            ---
+            kind: ImageStream
+            apiVersion: image.openshift.io/v1
+            metadata:
+              annotations:
+                opendatahub.io/notebook-image-name: "$(params.name)"
+                opendatahub.io/notebook-image-desc: "$(params.description)"
+                opendatahub.io/notebook-image-creator: "$(params.creator)"
+                opendatahub.io/notebook-image-url: "$(tasks.get-base-image.results.baseImage)"
+              name: "$(context.pipelineRun.name)"
+              namespace: "$(context.pipelineRun.namespace)"
+              labels:
+                opendatahub.io/notebook-image: 'true'
+            spec:
+              lookupPolicy:
+                local: true
+              tags:
+                - name: latest
+            EOM
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: buildah-requirements
+spec:
+  params:
+    - name: BASE_IMAGE
+      type: string
+    - description: Reference of the image buildah will produce.
+      name: IMAGE
+      type: string
+    - default: >-
+        registry.redhat.io/rhel8/buildah@sha256:6a68ece207bc5fd8db2dd5cc2d0b53136236fb5178eb5b71eebe5d07a3c33d13
+      description: The location of the buildah builder image.
+      name: BUILDER_IMAGE
+      type: string
+    - default: vfs
+      description: Set buildah storage driver
+      name: STORAGE_DRIVER
+      type: string
+    - default: "true"
+      description: >-
+        Verify the TLS on the registry endpoint (for push/pull to a non-TLS
+        registry)
+      name: TLSVERIFY
+      type: string
+    - default: oci
+      description: "The format of the built container, oci or docker"
+      name: FORMAT
+      type: string
+  workspaces:
+    - name: requirements
+      readonly: true
+      description: the list of pinned python package we install in the produced image
+  volumes:
+    - name: /var/lib/containers
+      emptyDir: {}
+  stepTemplate:
+      image: $(params.BUILDER_IMAGE)
+      env:
+        - name: STORAGE_DRIVER
+          value: $(params.STORAGE_DRIVER)
+        - name: BUILDAH_FORMAT
+          value: $(params.FORMAT)
+      volumeMounts:
+        - name: containers
+          mountPath: /var/lib/containers
+  steps:
+    - args: ["buildah", "from", "--name", "cnbi-image", "--tls-verify=$(params.TLSVERIFY)", "docker://$(params.BASE_IMAGE))"]
+    - args: ["buildah", "copy", "cnbi-image", "$(worspaces.requirements.path)/requirements-pinned.txt", "/tmp/"]
+    - args: ["buildah", "run", "cnbi-image", "--", "/usr/bin/pip", "install", "-r", "/tmp/requirements-pinned.txt"]
+    - args: ["buildah", "commit", "cnbi-image", "$(params.IMAGE)"]
+    - args: ["buildah", "push", "--tls-verify=$(params.TLSVERIFY)", "$(params.IMAGE)"]
```

Comment:
maybe we can use the generate-jupyterhub.yaml from PR: #108 
and extend it a little bit more to have this flow.
in this way we can use the ClusterTask buildah ?
wdyt?

Reply:
I think this clean for now, maybe after mvp .
We can refactor.